spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/autoconfig/AxonServerBusAutoConfiguration.java

```diff
@@ -0,0 +1,142 @@
+import org.springframework.context.annotation.Primary;
+
+/**
+ * Configures Axon Server as implementation for the CommandBus, QueryBus and EventStore.
+ *
+ * @author Marc Gathier
+ * @since 4.0
+ */
+@Configuration
+@AutoConfigureAfter(AxonServerAutoConfiguration.class)
+@AutoConfigureBefore(AxonAutoConfiguration.class)
+@ConditionalOnClass(AxonServerConfiguration.class)
+@ConditionalOnProperty(name = "axon.axonserver.enabled", matchIfMissing = true)
+public class AxonServerBusAutoConfiguration {
+
+    @Bean
+    @Primary
+    @ConditionalOnMissingQualifiedBean(qualifier = "!localSegment", beanClass = CommandBus.class)
+    public AxonServerCommandBus axonServerCommandBus(AxonServerConnectionManager axonServerConnectionManager,
+                                                     AxonServerConfiguration axonServerConfiguration,
+                                                     @Qualifier("localSegment") CommandBus localSegment,
+                                                     @Qualifier("messageSerializer") Serializer messageSerializer,
+                                                     RoutingStrategy routingStrategy,
+                                                     CommandPriorityCalculator priorityCalculator,
+                                                     CommandLoadFactorProvider loadFactorProvider,
+                                                     TargetContextResolver<? super CommandMessage<?>> targetContextResolver) {
+        return AxonServerCommandBus.builder()
+                .axonServerConnectionManager(axonServerConnectionManager)
+                .configuration(axonServerConfiguration)
+                .localSegment(localSegment)
+                .serializer(messageSerializer)
+                .routingStrategy(routingStrategy)
+                .priorityCalculator(priorityCalculator)
+                .loadFactorProvider(loadFactorProvider)
+                .targetContextResolver(targetContextResolver)
+                .build();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(QueryBus.class)
+    public AxonServerQueryBus queryBus(AxonServerConnectionManager axonServerConnectionManager,
+                                       AxonServerConfiguration axonServerConfiguration,
+                                       AxonConfiguration axonConfiguration,
+                                       TransactionManager txManager,
+                                       @Qualifier("messageSerializer") Serializer messageSerializer,
+                                       Serializer genericSerializer,
+                                       QueryPriorityCalculator priorityCalculator,
+                                       QueryInvocationErrorHandler queryInvocationErrorHandler,
+                                       TargetContextResolver<? super QueryMessage<?, ?>> targetContextResolver) {
+        SimpleQueryBus simpleQueryBus =
+                SimpleQueryBus.builder()
```

Comment:
Nit: I'd expect `.messageMonitor(axonConfiguration.messageMonitor(QueryBus.class, "queryBus"))` to indent to ` SimpleQueryBus.`.